R-packages/evalcast/R/geo_type_selector.R

```diff
@@ -0,0 +1,50 @@
+
+geo_type_selector <- function(geo_type, predictions_cards) {
+  # tries to determine which geo_type we want and/or validate
+  # our selection based on the content of predictions_cards
+  # 
+  # Wraps the function below for ease of testing (we test the validator,
+  # rather than this function)
+  pred_card_types <- predictions_cards %>%
+    transmute(type_len = nchar(.data$geo_value),
+              has_chars = str_detect(.data$geo_value, "[aA-zZ]")) %>%
+    distinct()
+  geo_type <- match.arg(geo_type, c("county", "hrr", "msa", "dma", "state"))
+  validate_geo_type(geo_type, pred_card_types)
+}
+
+validate_geo_type <- function(geo_type, pred_card_types) {
+  if (nrow(pred_card_types) > 1L) {
+    assert_that(geo_type %in% c("hrr", "msa", "dma"),
+                msg = paste("predictions_cards",
+                            "contain multiple geo_types."))
+  }
+  if (any(pred_card_types$has_chars)) {
+    assert_that(geo_type %in% c("state","county"),
+                msg = paste("predictions_cards contain non-numeric characters",
+                            "but geo_type was not `state`"))
+  }
+  if (geo_type == "state") {
```

Comment:
I don't understand why we want to try to automatically correct the geo type.

Reply:
I think we probably shouldn't, but we did previously. The new version *should* detect state vs. county automatically (the old version did this), but also allow geo_type as an argument. The old version didn't have the argument, so we would break a number of things if we don't keep the auto-detection.